### PR TITLE
Add docs for the doctrine/persistence project.

### PR DESCRIPTION
**Pull request by jwage**

### CODE REVIEW

File: docs/en/index.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+The Doctrine Persistence documentation is a reference guide to everything you need
+to know about the project.
+
+Getting Help
+------------
+
+If this documentation is not helping to answer questions you have about the
+Doctrine Persistence project, don't panic. You can get help from different sources:
+
+-  Gitter chat room `#doctrine/event-manager <https://gitter.im/doctrine/persistence>`_
```
Comment (malarzm) [inline docs/en/index.rst]: You missed `event-manager` in room's name ;)

File: docs/en/index.rst [hunk 2]
```diff
@@ -0,0 +1,22 @@
+Persistence Documentation
+===========================
```
Comment (Majkl578) [inline docs/en/index.rst]: underline is longer than the title :innocent:

File: docs/en/reference/index.rst [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+A Doctrine object mapper looks like this when implemented.
+
+.. code-block:: php
+
+    class User
+    {
+        /** @var string */
+        private $username = '';
+
+        public function __construct(string $username) : void
```
Comment (Majkl578) [inline docs/en/reference/index.rst]: constructor can't be void (can't have a return type)

File: docs/en/reference/index.rst [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+was extracted and eventually moved to this standalone project.
+
+A Doctrine object mapper looks like this when implemented.
+
+.. code-block:: php
+
+    class User
+    {
+        /** @var string */
+        private $username = '';
```
Comment (Majkl578) [inline docs/en/reference/index.rst]: please drop default value :)

**Pull request closed without merging**